### PR TITLE
feat: Language in config file

### PR DESCRIPTION
**Pull request by buhtz**
New key "global.language" in config file to customize the UI language.

Language codes are used as values. If empty (or key not present) the systems current locale is used. If "en" or unknown the sources strings (in English) are used as fallback.

No matter that it hurt me I decided against creating a unit test for "global.language" config value. I have a refactoring about the config-file handling in my mind which will ease up a lot of things including unit testing.

### CODE REVIEW

File: common/config.py [hunk 1]
```diff
@@ -192,6 +192,9 @@ class Config(configfile.ConfigFileWithProfiles):
     PLUGIN_MANAGER = pluginmanager.PluginManager()
 
     def __init__(self, config_path=None, data_path=None):
+        # Note: The main profiles name here is translated using the systems
+        # current locale because the language code in the config file wasn't
+        # read yet.
         configfile.ConfigFileWithProfiles.__init__(self, _('Main profile'))
```
Comment (buhtz) [inline common/config.py:198]: In the future we should remove the ability to translate the first profiles name. A name as an identifier shouldn't be translated at all. Related to #1371

File: common/config.py [hunk 2]
```diff
@@ -326,7 +329,10 @@ def __init__(self, config_path=None, data_path=None):
         self.inhibitCookie = None
         self.setupUdev = tools.SetupUdev()
 
-        tools.initiate_translation()
+        tools.initiate_translation(self.language())
+
+        # Workaround
+        self.default_profile_name = _('Main profile')
```
Comment (buhtz) [inline common/config.py:335]: Here the main profiles name is translated again now using the correct configured language. A bit dump I know. 😄

File: common/config.py (hunk 1 above)
Comment (aryoda) [reply common/config.py:198]: or even better: Why do we need a default profile at all?

**Comment by aryoda** [comment]
I did not yet deep dive into the change set but AFAIR changing the config file format (here: by adding a new entry) requires to increment the config file version in `common/config.py` (CONFIG_VERSION) so that BiT automatically updates older config files to the new format.

I couldn't see this in the change set (and this also possibly requires to update the config file format of unit test configs to avoid an unexpected output that fails some unit tests)...

**Comment by buhtz** [comment]
> but AFAIR changing the config file format (here: by adding a new entry) requires to increment the config file version

I'm aware of that version variable but I I don't see a need to increment it. Adding a new variable do not "change the format". It do not break existing behavior. And the variable do not need to be present in the config file because there is a default value for it.

Technically there is no need to update old config files if you do not modify the name or format of existing variables. If the line `global.language` is not present the default value is used. If the value is stored the first time it is just added to the config file. This is quit flexible.

**Comment by aryoda** [comment]
> Adding a new variable do not "change the format"

I think reusing the new config format with an older BiT version will no longer work (though I am not sure if this is a use case).

Anyhow I think it would be more clean to increment the config file version every time we change the format (even though it is non-destructive/additive).

I am not sure how Germar handled this in the past, perhaps browsing the change history of the version could help.

**Comment by buhtz** [comment]
> I think reusing the new config format with an older BiT version will no longer work (though I am not sure if this is a use case).

It is an unusual case but possible. I'll test that. But I'm quit deep into the code of `config.py` and `configfile.py`. In my understanding this won't be a problem. The class read only what it needs and ignores the rest.

> Anyhow I think it would be more clean to increment the config file version every time we change the format (even though it is non-destructive/additive).

I don't think it is clean to even have a version number for config formats. That is my problem with the solution. I don't see a need for such a version number. There are other ways to handle modifications of variable names and similar things in a config file.

**Comment by buhtz** [comment]
> I'll test that.

I added two lines to my BIT 1.3.3 config file.

```
foo.bar=8
global.bähm=nein
```

No problem. I can read all snapshot profiles and also modify them. That two lines are still in the config file.

**Review by aryoda — APPROVED** [review]

**Pull request merged**